common/metrics/tally/statsd/reporter.go

```diff
@@ -112,3 +111,36 @@ func (r *temporalTallyStatsdReporter) Capabilities() tally.Capabilities {
 func (r *temporalTallyStatsdReporter) Flush() {
 	r.tallystatsd.Flush()
 }
+
+// embedTags adds the sorted list of tags directly in the stat name.
+// For example, if the stat is `hello.world` and the tags are `{universe: milkyWay, planet: earth}`,
+// the stat will be emitted as `hello.world.planet.earth.universe.milkyWay`.
+func embedTags(name string, tags map[string]string) string {
+	// Sort tags so they are in a consistent order when emitted.
+	var keys []string
+	for k := range tags {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	var buffer strings.Builder
+	buffer.WriteString(name)
+	for _, tk := range keys {
+		// adding "." as delimiter so that it will show as different parts in Graphite/Grafana
+		buffer.WriteString("." + tk + "." + tags[tk])
+	}
+
+	return buffer.String()
+}
+
+// appendSeparatedTags adds the sorted list of tags using the DogStatsd/InfluxDB supported tagging protocol.
+// For example, if the stat is `hello.world` and the tags are `{universe: milkyWay, planet: earth}` and the separator is `,`,
+// the stat will be emitted as `hello.world,planet=earth,universe=milkyWay`.
+func appendSeparatedTags(name string, separator string, tags map[string]string) string {
```

Comment:
> It looks like embedTags and appendSeparatedTags could be merged. This would result in tags having the same sort order.

Only embedded tags need to be sorted, that's why I separated them. The order of tags does not matter if they are emitted as a serialized map of tags and not embedded in the stat name itself.

> 
> Declare `.` and `=` as package level constants.

Sure I can do that. Can you give some additional insight into the benefit of that though?

> Is there a link to the spec backing this statement?
> 
> > using the DogStatsd/InfluxDB supported tagging protocol.

Available information on the tagging protocol is linked in the description.

Reply:
Done.